tests/test_iv.py

```diff
@@ -1,31 +1,215 @@
 import numpy as np
+import pandas as pd
 import pytest
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.packages import importr
 
 from pyfixest.estimation.estimation import feols
 from pyfixest.utils.utils import get_data
 
+# Activate pandas2ri to enable conversion between pandas DataFrames and R DataFrames
+pandas2ri.activate()
+
+# Import the ivDiag package
+ivDiag = importr("ivDiag")
+
+
+# Enable the automatic conversion between pandas DataFrame and R DataFrame
+pandas2ri.activate()
+
+
+@pytest.fixture(scope="module")
+def r_results():
+    np.random.seed(1)
+
+    # Number of observations
+    n = 500
+
+    # Simulate the data
+    # Instrumental variable
+    z = np.random.binomial(1, 0.5, size=n)
+    z2 = np.random.binomial(1, 0.5, size=n)
+    # Endogenous variable
+    d = 0.5 * z + 1.5 * z2 + np.random.normal(size=n)
+
+    # Control variables
+    c1 = np.random.normal(size=n)
+    c2 = np.random.normal(size=n)
+
+    # Outcome variable
+    y = 1.0 + 1.5 * d + 0.8 * c1 + 0.5 * c2 + np.random.normal(size=n)
+
+    # Cluster variable
+    cluster = np.random.randint(1, 50, size=n)
+
+    # Sampling weights (random uniform distribution between 1 and 3 for example)
+    weights = np.random.uniform(1, 3, size=n)
+
+    # Create a DataFrame
+    data = pd.DataFrame(
+        {
+            "d": d,
+            "y": y,
+            "z": z,
+            "z2": z2,
+            "c1": c1,
+            "c2": c2,
+            "cluster": cluster,
+            "weights": weights,
+        }
+    )
+
+    # Convert the DataFrame to an R DataFrame
+    data_r = pandas2ri.py2rpy(data)
+
+    # Define the variables
+    Y = "y"
+    D = "d"  # Endogenous treatment
+    Z = "z"  # Instrumental variable
+    controls = ["c1", "c2"]  # Covariates of control variables
+    cl = "cluster"
+
+    # Convert the variables and controls to R objects
+    Y_r = ro.StrVector([Y])
+    D_r = ro.StrVector([D])
+    Z_r = ro.StrVector([Z])
+    controls_r = ro.StrVector(controls)
+    cl_r = ro.StrVector([cl])
+
+    # Call the ivDiag function from the ivDiag package
+    result_with_weights = ivDiag.ivDiag(
+        Y=Y_r,
+        D=D_r,
+        Z=Z_r,
+        controls=controls_r,
+        data=data_r,
+        weights="weights",
+        cl=cl_r,
+    )
+
+    result_without_weights = ivDiag.ivDiag(
+        Y=Y_r, D=D_r, Z=Z_r, controls=controls_r, data=data_r, cl=cl_r
+    )
+
+    return {
+        "with_weights": result_with_weights.rx2("F_stat"),
+        "without_weights": result_without_weights.rx2("F_stat"),
+        "data": data,
+    }
+
+
+@pytest.mark.parametrize("has_weight", [0.0, 1.0])
```

Comment:
Maybe annoyingly nit-picky (sorry about that! 😅 ): 

I would rename the test parametrization to 
```python
@pytest.mark.parametrize("has_weight", [False, True])
@pytest.mark.parametrize("adj_vcov", ["iid", "hetero", {"CRV1":"cluster"}])
```
as it is easier to read?

Reply:
This is a bad habit of mine that I always tend to work on numeric value ;). I will fix this!